CONTRIBUTING.md

```diff
@@ -1,3 +1,456 @@
+Ensure each of your commits is signed-off by using `git commit -s`. Function Clarity
+members highly values readable, idiomatic Go code. Familiarize yourself with
+the [Coding Style](#coding-style) section below and try to preempt any comments
+your reviewers would otherwise leave. Most repos will contain `Makefile`s or
+[`mage`][mage] targets that will help you build/test/lint/verify your changes.
+If you don't see a target for your use case, maybe that could be your first
+contribution!
+
+<!--
+TODO(contribute): Decide on stance around testing libraries: https://github.com/golang/go/wiki/TestComments#assert-libraries
+-->
+
+All code should be covered by tests. We strive, wherever possible, to follow
+the common Go [test review comments]. We encourage the use of table-driven
+unit tests - you can find an example [below](#prefer-table-driven-tests). Note
+that when opening a PR your reviewer will expect you to detail how you've
+tested your work. For all but the smallest changes, some manual testing is
+encouraged in addition to unit tests.
+
+All documentation is under revision control. Any change that introduces new
+behavior or changes existing behavior must include updates to any relevant
+documentation. Please keep documentation changes in distinct commits.
+
+Once your change is written, tested, and documented the final step is to have it
+reviewed! You'll be presented with a template and a small checklist when you
+open a PR. Please read the template and fill out the checklist. Please make all
+requested changes in subsequent commits. This allows your reviewers to see what
+has changed as you address their comments. Be mindful of your commit history as
+you do this - avoid commit messages like "Address review feedback" if possible.
+If doing so is difficult a good alternative is to rewrite your commit history
+to clean them up after your PR is approved but before it is merged.
+
+In summary, please:
+
+- Discuss your change in a GitHub issue before you start.
+- Use your Git commit messages to communicate your intent to your reviewers.
+- Sign-off on all git commits by running `git commit -s`
+- Add or update tests for all changes.
+- Preempt [coding style](#coding-style) review comments.
+- Update all relevant documentation.
+- Don't force push to address review feedback. Your commits should tell a story.
+- If necessary, tidy up your git commit history once your PR is approved.
+
+Thank you for reading through our contributing guide! We appreciate you taking
+the time to ensure your contributions are high quality and easy for our
+community to review and accept. Please don't hesitate to [reach out to
+us][contact] if you have any questions about contributing!
+
+## Coding Style
+
+Function Clarity prefers not to maintain its own style guide, but we do
+enforce the style and best practices established by the Go project and its
+community. This means contributors should:
+
+- Follow the guidelines set out by the [Effective Go] document.
```

Comment:
Review the links throughout the doc, as many are probably broken.
You'll want to include the section at the bottom of the reference contribution guide to fix these: https://github.com/kubernetes/sig-release/blob/2475370b19c621dc199ba96d2151614f932cd009/CONTRIBUTING.md?plain=1#L591-L608